extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/MSQTaskQueryMaker.java

```diff
@@ -283,6 +283,10 @@ public QueryResponse<Object[]> runQuery(final DruidQuery druidQuery)
 
     MSQTaskQueryMakerUtils.validateRealtimeReindex(querySpec);
 
+    final Map<String, Object> context = new HashMap<>();
+    context.put(PlannerContext.CTX_LOOKUP_LOADING_MODE, plannerContext.getLookupLoadingSpec().getMode());
+    context.put(PlannerContext.CTX_LOOKUPS_TO_LOAD, plannerContext.getLookupLoadingSpec().getLookupsToLoad());
```

Comment:
We should only set this if the mode is ONLY_REQUIRED no ?

Reply:
The logic in `MSQWorkerTask#getLookupLoadingSpec` loads all lookups if these keys aren't present in the context (for backward compatibility reasons). Hence, we need to set these explicitly even for not loading any lookups.

Attaching snippet of the code in `MSQWorkerTask#getLookupLoadingSpec`:
```java
    final Object lookupModeValue = getContext().get(PlannerContext.CTX_LOOKUP_LOADING_MODE);
    if (lookupModeValue == null) {
      return LookupLoadingSpec.ALL;
    }
```

Reply:
+1 to what @cryptoe said.

@Akshat-Jain , @cryptoe meant that we shouldn't add the CTX_LOOKUPS_TO_LOAD key if the value is going to be null anyway. The key CTX_LOOKUP_LOADING_MODE should always be added.

Reply:
@kfaraz Oh I see, makes sense. Thanks, I misinterpreted the comment.

Reply:
Have made the change.